plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/widget/PasteFromClipboardDataCommandHandler.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Martin Erich Jobst
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Martin Jobst - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.ui.widget;
+
+import java.util.Objects;
+
+import org.eclipse.nebula.widgets.nattable.command.AbstractLayerCommandHandler;
+import org.eclipse.nebula.widgets.nattable.coordinate.PositionCoordinate;
+import org.eclipse.nebula.widgets.nattable.copy.command.PasteDataCommand;
+import org.eclipse.nebula.widgets.nattable.edit.command.EditUtils;
+import org.eclipse.nebula.widgets.nattable.edit.command.UpdateDataCommand;
+import org.eclipse.nebula.widgets.nattable.layer.IUniqueIndexLayer;
+import org.eclipse.nebula.widgets.nattable.layer.LayerUtil;
+import org.eclipse.nebula.widgets.nattable.layer.cell.ILayerCell;
+import org.eclipse.nebula.widgets.nattable.selection.SelectionLayer;
+import org.eclipse.swt.dnd.Clipboard;
+import org.eclipse.swt.dnd.TextTransfer;
+import org.eclipse.swt.widgets.Display;
+
+public class PasteFromClipboardDataCommandHandler extends AbstractLayerCommandHandler<PasteDataCommand> {
```

Comment:
We have a org.eclipse.fordiac.ide.ui.FordiacClipboard . This is used to keep graphical elements and textual ones.
Example: when copying an edit part in the editor, it does not delete a previously copied text in the clipboard. 

I am not really sure what you are doing here, so i don't know if that is relevant, but i just wanted to mention that this functionality exits.

Reply:
Thank you for letting me know about the `FordiacClipboard`. The helper methods for handling textual contents have been removed from the `FordiacClipboard` in https://github.com/eclipse-4diac/4diac-ide/commit/832e1af821a6a39daa9a743b1b40a9f7b28dcfaa, so I am using the OS clipboard directly.